temporalio/envconfig.py

```diff
@@ -0,0 +1,342 @@
+"""Environment and file-based configuration for Temporal clients.
+
+This module provides utilities to load Temporal client configuration from TOML files
+and environment variables.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Any, Mapping, Optional, Union
+
+from typing_extensions import TypeAlias
+
+from temporalio.bridge.temporal_sdk_bridge import envconfig as _bridge_envconfig
+from temporalio.service import ConnectConfig, TLSConfig
+
+DataSource: TypeAlias = Union[
+    str, bytes
+]  # str represents a file path, bytes represents raw data
```

Comment:
This is a bit confusing. It is very normal to provide PEM file contents as a string in languages. We should differentiate file path better. Maybe `Union[pathlib.Path, str, bytes]` and only paths are treated as such?

Reply:
I've changed the typing to your suggestion but I'm unsure whether or not to keep the `str` type.

The core TOML parser reads the cert configs into:
```
pub enum DataSource {
    Path(String),
    Data(Vec<u8>),
}
```
so lang only sees these cert configs as bytes or a path (which we now read as an actual `Path` type).

Users can provide file contents as string but it would only be possible programmatically, like the (added) test `test_read_source_from_string_content`. I am not sure if we wanted envconfig to be used this way

Reply:
`test_read_source_from_string_content` reads in string content, but asserts that it is read correctly into bytes.
So it's still _from_ string content so to speak.